assets/js/script.js

```diff
@@ -0,0 +1,270 @@
+  setOnclickAttribute(firstButton, numbers.ONE);
+  sectionPagination.appendChild(firstButton);
+
+  for (let index = numbers.TWO; index <= totalPage; index++) {
+    const pageButton = createHtmlElement(
+      pageStrings.BUTTON,
+      index,
+      null,
+      `btn-${index}`
+    );
+    setOnclickAttribute(pageButton, index);
+    sectionPagination.appendChild(pageButton);
+  }
+
+  const pageNextButton = createHtmlElement(
+    pageStrings.BUTTON,
+    pageStrings.NEXT,
+    null,
+    pageStrings.NEXT_BUTTON
+  );
+  setOnclickAttribute(pageNextButton);
+  sectionPagination.appendChild(pageNextButton);
+  sethtmlQuerySelectorObject();
+};
+
+const paginate = async (page) => {
+  if (isCurrentPagePrevious(page)) pageStack.pop();
+  else if (isCurrentPageNext(page)) pageStack.push(++currentPage);
+  else pageStack.push(page);
+
+  currentPage = pageStack[pageStack.length - numbers.ONE];
+  setVideos();
+  toggleActiveButton();
+  toggleButton();
+};
+
+const createHtmlElement = (tagName, text, className, id) => {
+  const htmlTag = document.createElement(tagName);
+  text && (htmlTag.innerText = text);
+  id && (htmlTag.id = id);
+  className && htmlTag.classList.add(className);
+  return htmlTag;
+};
+
+const setOnclickAttribute = (button, text = pageStrings.NEXT) => {
+  button.addEventListener("click", function () {
+    paginate(
+      text == pageStrings.NEXT
+        ? pageStrings.NEXT
+        : text == pageStrings.PREV
+        ? pageStrings.PREV
+        : text,
+      this
+    );
+  });
+};
+
+const sethtmlQuerySelectorObject = () => {
+  htmlQuerySelectorObject.previousButton = document.querySelector(
+    `#${pageStrings.PREVIOUS_BUTTON}`
+  );
+  htmlQuerySelectorObject.nextButton = document.querySelector(
+    `#${pageStrings.NEXT_BUTTON}`
+  );
+  htmlQuerySelectorObject.activeButton = document.querySelector(
+    `.${pageStrings.ACTIVE_BUTTON}`
+  );
+};
+
+const toggleActiveButton = () => {
+  let { activeButton: currentActiveButton } = htmlQuerySelectorObject;
+  currentActiveButton.classList.remove(pageStrings.ACTIVE_BUTTON);
+  currentActiveButton = document.querySelector(`#btn-${currentPage}`);
+  currentActiveButton.classList.add(pageStrings.ACTIVE_BUTTON);
+  htmlQuerySelectorObject.activeButton = currentActiveButton;
+};
+
+const toggleButton = () => {
+  togglePreviousButton();
+  toggleNextButton();
+};
+
+const togglePreviousButton = () => {
+  const { previousButton } = htmlQuerySelectorObject;
+  isCurrentPageEquals(numbers.ONE)
+    ? (previousButton.disabled = true)
+    : (previousButton.disabled = false);
+};
+
+const toggleNextButton = () => {
+  const { nextButton } = htmlQuerySelectorObject;
+  isCurrentPageEquals(totalPage)
+    ? (nextButton.disabled = true)
+    : (nextButton.disabled = false);
+};
+
+const isCurrentPageEquals = (pageCount) => currentPage == pageCount;
+const isCurrentPagePrevious = (page) => page == pageStrings.PREV;
+const isCurrentPageNext = (page) => page == pageStrings.NEXT;
```

Comment:
I think we should see this differently like -
hasCurrentPage;
hasPrevPage;
hasNextPage;

Please think about this.